### PR TITLE
Remove guava and guice usage from the plugin

### LINKED ISSUE #1593: remove the use of Guava (and Guice) from the plugin
### Your checklist for this issue

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Link to any upstream changes that might be required (for example Jenkins Core pull request)

### Feature Request

There is work underway in Jenkins to upgrade Guava in Jenkins core.  Newer versions of Guava than the one that are used in Jenkins are not binary compatible.
Whilst the plugin is not identified as using any breaking API we also want to not expose every library from Jenkins to plugins and so by removing Guava and Guice where it is not needed or replacing it with something else will better enable this work.

### PR DESCRIPTION
**Pull request by jtnord**
Remove the use of Guava and Guice from the plugin in order to be compatible with all future Jenkins versions where Guava may be bumped or not exposed.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

fixes #1593

### CODE REVIEW

**Review by jetersen — APPROVED** [review]
LGTM

**Comment by jtnord** [comment]
I was a little scared about the removal of Guice in 5a63993facbe5c75df660578183d64424205af6e however all the integration tests pass.  is there any other testing that you believe I should do in this area?

**Comment by jetersen** [comment]
@jtnord not to my knowledge we have pretty decent coverage on the important bits.

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java [hunk 1]
```diff
@@ -164,7 +163,7 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
                                 new Object[]{target, names[i], t == Secret.class || Attribute.calculateIfSecret(target, names[i]) ? "****" : value});
                     }
                 } else if (t.isPrimitive()) {
-                    args[i] = defaultValue(t);
+                    args[i] = defaultPrimativeValue(t);
```
Comment (timja) [inline plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java:166]: https://en.wiktionary.org/wiki/primative


```suggestion
                    args[i] = defaultPrimitiveValue(t);
```

**Review by timja — CHANGES_REQUESTED** [review]
typoes need fixing, rest look good

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java (hunk 1 above)
Comment (jtnord) [reply plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java]: fixed in 95befad5de4bf6c72632f504c62838a7c3757c3a

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Do you need a release?

**Comment by jtnord** [comment]
> Do you need a release?

would be helpful if you can.

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/DefaultConfiguratorRegistry.java [hunk 2]
```diff
@@ -80,21 +79,15 @@ public Configurator lookupOrFail(Type type) throws ConfiguratorException {
...
-            return cache.get(type);
-        } catch (ExecutionException e) {
-            return null;
-        }
+        return cache.get(type);
     }
 
-    private LoadingCache<Type, Configurator> cache = CacheBuilder.newBuilder()
-            .expireAfterAccess(10, TimeUnit.SECONDS)
+    private LoadingCache<Type, Configurator> cache = Caffeine.newBuilder()
```
Comment (jglick) [inline plugin/src/main/java/io/jenkins/plugins/casc/impl/DefaultConfiguratorRegistry.java:85]: #1595

**Pull request merged**